src/celengine/astroobj.cpp

```diff
@@ -5,18 +5,61 @@
 #include "astroobj.h"
 #include "category.h"
 
+AstroObject::AstroObject(AstroObject &o)
+{
+    setIndex(o.getIndex());
+
+    auto cats = o.getCategories();
+    if (cats != nullptr)
+        for (auto *cat : *cats)
+            addToCategory(cat);
+}
+
+AstroObject::AstroObject(AstroObject &&o)
+{
+    setIndex(o.getIndex());
+
+    auto cats = o.getCategories();
+    if (cats != nullptr)
```

Comment:
and this block is redundant, all you need is `m_categories = o.m_categories; o.m_categories = nullptr;`

Reply:
Not exactly. Categories would't know the object is moved. This would lead to inconsistence.

Reply:
in that case why not to remove reference to old object and add one to new object?

Reply:
Hm... It could be a bit more efficient, but modification of elements inside set will change its order, hence will make it decoherent. It could have a point for map or vector, but is it worth effort?